README.md

```diff
@@ -1,4 +1,1060 @@
+```bash
+sudo launchctl stop org.nixos.nix-daemon
+sudo launchctl start org.nixos.nix-daemon
+```
+Leave `allow-import-from-derivation` set to `true` for `haskell.nix` for work correctly.
+
+## 3.2. `nix/iogx-config.nix`
+
+```nix
+# Example of a valid nix/iogx-config.nix
+{ 
+  repoRoot = ../.; 
+  systems = [ "x86_64-linux" "x86_64-darwin" ]; 
+  haskellCompilers = [ "ghc8107" ]; 
+  defaultHaskellCompiler = "ghc8107"; 
+  shouldCrossCompile = true; 
+} 
+```
+
+This file must exist and it contains the main configuration values for your project.
+
+### 3.2.1. `repoRoot` 
+
+A Nix path to an existing folder containing your `cabal.project` file.
+
+In future versions this field will be removed and it will default to the repository top-level directory, but until then it must be explicitly set (usually to `../.`).
+
+### 3.2.2. `systems`
+
+The non-empty list of systems against which your project can be built. 
+
+These are standard Nix values found in `nixpkgs.stdenv.system`.
+
+This field affects your final [flake outputs](#310-flake-outputs-format).
+
+This field is required.
+
+### 3.2.3. `haskellCompilers`
+
+The non-empty list of GHC versions that can build your project. 
+
+Currently two GHC versions are supported and provided by IOGX: `ghc8107` and `ghc927`.
+
+This field affects your final [flake outputs](#310-flake-outputs-format).
+
+This field is required.
+
+### 3.2.4. `defaultHaskellCompiler`
+
+Only one compiler at a time can be visible in the `$PATH` in the shell.
+
+When calling `nix develop`, the `defaultHaskellCompiler` will be selected. 
+
+To enter a shell with a different compiler, see to the [flake outputs](#310-flake-outputs-format) section. 
+
+This field is optional and defaults to the first (leftmost)  compiler in [`haskellCompilers`](#323-haskellcompilers).
+
+### 3.2.5. `shouldCrossCompile`
+
+Cross-compilation on Windows is available via `mingwW64` on `x86_64-linux` only. 
+
+If your project cannot be cross-compiled then set this field to `false`. 
+
+This field affects your final [flake outputs](#310-flake-outputs-format).
+
+This field is optional and defaults to `true`.
+
+## 3.3. `nix/haskell-project.nix`
+
+```nix
+# Default nix/haskell-project.nix
+{ inputs 
+, inputs' 
+, pkgs 
+, { haskellCompiler
+  , enableHaddock 
+  , enableProfiling 
+  , enableCross 
+  }@meta 
+}:
+{
+  cabalProjectLocal = ""; 
+  sha256map = { }; 
+  shellWithHoogle = false; 
+  modules = [ ]; 
+  overlays = [ ];
+}
+```
+
+This file describes your Haskell project and will be used internally to call `haskell.nix:cabalProject'`.
+
+This file will be evaluated once for each element in your GHC build matrix: if your [`haskellCompilers`](#323-haskellcompilers) has 2 elements, and if [`shouldCrossCompile`](#325-shouldcrosscompile) is set to `true`, then this file will be called 8 times (taking into account profiled and non-profiled builds).
+
+If this file does not exist then it will default to an empty project.
```

Comment:
What is an "empty project"? One with no Haskell packages? Or one that just takes the `cabal.project` in the root as given with no special settings?